internal/isoeditor/rhcos.go

```diff
@@ -6,66 +6,94 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"sync"
+	"regexp"
 
-	"github.com/openshift/assisted-service/pkg/s3wrapper"
+	"github.com/openshift/assisted-service/internal/isoutil"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// CustomIsoTempDir is a temporary directory pattern for the custom ISO
+	CustomIsoTempDir string = "customiso"
+	// BaseIsoTempDir is a temporary directory pattern for the extracted base ISO
+	BaseIsoTempDir string = "baseiso"
+	// CustomIsoObjectName is a name pattern for the ISO template
+	CustomIsoObjectName string = "rhcos-%s-minimal.iso"
 )
 
 type Editor interface {
-	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error
+	CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) (string, error)
 }
 
 type rhcosEditor struct {
-	isoHandler    Handler
-	objectHandler s3wrapper.API
+	isoHandler    isoutil.Handler
 	rootfsURL     string
+	isoTmpWorkDir string
+	log           logrus.FieldLogger
 }
 
-func NewEditor(isoHandler Handler, objectHandler s3wrapper.API, rootfsURL string) Editor {
+func NewEditor(isoHandler isoutil.Handler, rootfsURL, isoTmpWorkDir string, log logrus.FieldLogger) Editor {
 	return &rhcosEditor{
-		isoHandler:    isoHandler,
-		objectHandler: objectHandler,
-		rootfsURL:     rootfsURL,
+		isoHandler: isoHandler,
+		rootfsURL:  rootfsURL,
+		log:        log,
 	}
 }
 
-func (e *rhcosEditor) CreateMinimalISOTemplate(ctx context.Context, openshiftVersion string) error {
+func CreateEditor(isoPath string, rootfsURL string, log logrus.FieldLogger) Editor {
+	isoTmpWorkDir, err := ioutil.TempDir("", BaseIsoTempDir)
```

Comment:
I wasn't sure if I wanted to do this because it's unclear whose responsibility it is to clean this directory up.

Now, if you use this function to get an editor instance, we probably need a public method to clean this up.
But if you use `NewEditor` then it's the caller's responsibility, right?

Maybe we should just have one way of doing this.

Reply:
Added a cleanup func to installerHandler for removing the temp folder, so no need to maintain BaseIsoTempDir now.